src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -108,6 +111,117 @@ const UserProfileModal: React.FunctionComponent<PropTypes> = ({
   );
 };
 
+const Badges: React.FC<{ user: WithId<User> }> = ({ user }) => {
+  useFirestoreConnect([
+    {
+      collection: "users",
+      doc: user.id,
+      subcollections: [{ collection: "visits" }],
+      storeAs: "userModal-visits",
+    },
+  ]);
+  const visits = useSelector(
+    (state) => state.firestore.ordered["userModal-visits"]
+  );
+  const venues = useSelector((state) => state.firestore.ordered.venues);
+
+  const playaTime = useMemo(() => {
+    if (!visits) return undefined;
+
+    const playaSeconds = visits.reduce((acc, v) => acc + v.timeSpent, 0);
+    const playaHours = Math.floor(playaSeconds / (60 * 60));
+    return playaHours > 1 ? `${playaHours}` : "> 1";
+  }, [visits]);
+
+  const venuesVisited = useMemo(() => {
+    if (!visits) return undefined;
+    return visits.filter((visit) => visit.id !== "Playa").length; // Playa does not count
+  }, [visits]);
+
+  const badges = useMemo(() => {
+    if (!visits || !venues) return [];
+    return [
+      ...visits
```

Comment:
no need to spread inside a new array, .filter and .map return a new array instance.